### DISCUSSION TITLE
touch1200 bootloader support in stock RP2040 images

### DISCUSSION THREAD

**Discussion by scruss** [micropython/micropython#11574] [RP2040 / Pico]
I know that USB handling in the rp2 port is very much a moving target with several groups working on refactoring it. If someone's working on this already and I've missed it, apologies.

The other ARM ports (stm32, samd) both appear to have touch1200 bootloader support baked in, but the rp2 uf2 images don't. This can cause problems for people who wish to try Arduino development on rp2, as Arduino relies on the running firmware having touch1200 support to update sketches. 

I realize that this is effectively helping other development frameworks out of a hole they're created, and also might affect a few people who have need to talk to a MicroPython board at 1200 baud.

An example of an Arduino user not being able to flash a board if a touch1200 firmware is missing: [Factory reset Pico no longer registers com port on windows - Raspberry Pi Forums](https://forums.raspberrypi.com/viewtopic.php?t=350680)

(touch1200 is a convention where development boards will drop into bootloader mode if their serial port is opened at 1200 baud. It avoids having to press buttons/disconnect when reprogramming)

**Comment by robert-hh** [comment]
touch1200 is available for the Arduino Nano RP2040 connect board. It can be enabled for the other boards as well by just adding the line:
`#define MICROPY_HW_USB_CDC_1200BPS_TOUCH (1)`
to their mpconfigboard.h files.
